packages/sanity/src/core/releases/contexts/upsell/ReleasesUpsellProvider.tsx

```diff
@@ -146,40 +145,91 @@ export function ReleasesUpsellProvider(props: {children: React.ReactNode}) {
     }
   }, [client, projectId])
 
-  const handleOpenDialog = useCallback(() => {
-    setUpsellDialogOpen(true)
+  const [releaseCount, setReleaseCount] = useState<number | null>(null)
 
-    telemetry.log(UpsellDialogViewed, {
-      feature: FEATURE,
-      type: 'modal',
-      source: 'navbar',
-    })
-  }, [telemetry])
+  const handleOpenDialog = useCallback(
+    (orgActiveReleaseCount?: number) => {
+      setUpsellDialogOpen(true)
+      if (orgActiveReleaseCount !== undefined) {
+        setReleaseCount(orgActiveReleaseCount)
+      }
+
+      telemetry.log(UpsellDialogViewed, {
+        feature: FEATURE,
+        type: 'modal',
+        source: 'navbar',
+      })
+    },
+    [telemetry],
+  )
+
+  const {releaseLimits$} = useReleaseLimits()
+  const {orgActiveReleaseCount$} = useOrgActiveReleaseCount()
 
   const guardWithReleaseLimitUpsell = useCallback(
-    (cb: () => void, throwError: boolean = false) => {
-      if (mode === 'default') {
-        return cb()
+    async (cb: () => void, throwError: boolean = false) => {
```

Comment:
the most minor of nits but naming wise I tend to not like having such truncated names and I know this is already from the previous code and not new _but_ just thought to bring it up: I imagine that it's "callback" but it took me a bit to read through and undestand it.